pkg/operator/controller/controller_dns_daemonset.go

```diff
@@ -194,6 +194,21 @@ func daemonsetConfigChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.
 		changed = true
 	}
 
+	// Detect changes to container commands
```

Comment:
The fact you are having to write this code is terrifying.  Is there a library-go function you should be using instead?

Reply:
Looks like library-go's resourceapply package might work.  However, the prospect of using library-go presents at least two salient challenges for us: (1) library-go is poorly documented and (2) library-go uses client-go whereas the DNS operator uses the controller-runtime client library.  Given these considerations, I don't see why using library-go would be less terrifying.